spec/policy/keycloak_role_check/keycloak_role_check_spec.lua

```diff
@@ -572,6 +574,81 @@ describe('Keycloak Role check policy', function()
           assert.not_same(ngx.status, 403)
         end)
       end)
+
+      describe("validate method roles", function()
+
+        before_each(function()
+          ngx.var = {
+            uri = '/foo'
+          }
+        end)
+
+        local context = {
+          jwt = {
+            realm_access = {
+              roles = { "known_role" }
+            },
+            resource_access = {
+              known_client = {
+                roles = { "role_of_known_client" }
+              }
+            }
+          },
+          service = {
+            auth_failed_status = 403,
+            error_auth_failed = "auth failed"
+          }
+        }
+
+        it("when jwt role matches and one method also match", function()
+          local role_check_policy = KeycloakRoleCheckPolicy.new({
+            scopes = {{
+                client_roles = { { name = "role_of_known_client", client = "known_client" } },
+                resource = "/foo",
+                methods = {"GET", "POST", "PUT"}
+              }}})
+          role_check_policy:access(context)
+          assert.not_same(ngx.status, 403)
```

Comment:
I'm not sure I understand this. By default this policy applies "whitelisting" instead of "blacklisting".
If I understood correctly the role matches and also the http method.
Why is the status a 4xx? What am I missing?

Reply:
On a related note, I think in these tests we should be explicit and mention whether they're whitelisting or blacklisting, and also add the attribute that sets that instead of relying on defaults. Then, we can have another test in a different section, if there is not one already, that checks that when the type is not set, it defaults to whitelisting.